isic_archive/models/image.py

```diff
@@ -29,16 +28,26 @@ def initialize(self):
 
         self.prefixSearchFields = ['lowerName', 'name']
 
+    @staticmethod
+    def generateIsicID():
+        taken = True
+        id_: int = 0
+
+        while taken:
+            id_ = secrets.randbelow(10_000_000)
+            taken = bool(Image().findOne({'name': f'ISIC_{id_}'}).count())
+
+        return f'ISIC_{id_}'
+
     def createEmptyImage(self, originalFileRelpath, parentFolder, creator, dataset,
                          batch):
-        newIsicId = Setting().get(PluginSettings.MAX_ISIC_ID) + 1
+        newIsicId = self.generateIsicID()
         image = self.createItem(
             name='ISIC_%07d' % newIsicId,
```

Comment:
Need to actually use the new value:

```suggestion
            name=newIsicId,
```